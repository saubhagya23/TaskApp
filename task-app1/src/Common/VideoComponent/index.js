import React, { Component } from 'react';

class VideoComponent extends Component {
  render(){
    const {
      videoURL,
      videoWidth,
      videoHeight,
    } = this.props;
    return(
      <div data-testid="video-comp-main-div">
        <video 
          width={(videoWidth !== null) ? videoWidth : '100%'} 
          height={(videoHeight !== null) ? videoHeight : '100%'} 
          autoPlay 
          muted
          data-testid="video-comp"
        >
          <source src={(videoURL !== '') ? videoURL : ''} type="video/mp4" data-testid="video-comp-source" />
          Your browser does not support the video tag.
        </video>
      </div>
    );
  }
}

export default VideoComponent;