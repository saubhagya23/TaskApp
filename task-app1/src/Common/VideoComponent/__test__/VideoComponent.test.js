import React from 'react';
import ReactDOM from 'react-dom';
import VideoComponent from './../index';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders input without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<VideoComponent/>, div)
})

it("renders video component correctly", () => {
  const { getByTestId } = render(<VideoComponent videoWidth="100%"/>)
  const videoCompSource = getByTestId('video-comp-source')
  expect(getByTestId('video-comp')).toContainElement(videoCompSource);
})

it("matches snapshot", () => {
  const videoCompTree = renderer.create(<VideoComponent videoWidth="100%" />).toJSON();
  expect(videoCompTree).toMatchSnapshot();
})
