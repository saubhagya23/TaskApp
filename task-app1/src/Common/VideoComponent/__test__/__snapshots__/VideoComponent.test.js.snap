exports[`test matches snapshot 1`] = `
<div
  data-testid="video-comp-main-div">
  <video
    autoPlay={true}
    data-testid="video-comp"
    height={undefined}
    muted={true}
    width="100%">
    <source
      data-testid="video-comp-source"
      src={undefined}
      type="video/mp4" />
    Your browser does not support the video tag.
  </video>
</div>
`;
